pkg/restapi/common/oidc/client.go

```diff
@@ -96,8 +86,8 @@ type Config struct {
 }
 
 // NewClient returns new client instance
-func NewClient(config *Config) *Client {
-	return &Client{
+func NewClient(config *Config) *OIDCClient {
+	return &OIDCClient{
 		provider:     config.Provider,
 		oauth2ConfigSupplier: func() oauth2Config {
```

Comment:
I see that this is just a refactor and the code already was there, but I am still curious why we have that config supplier instead of just passing the normal object - what is the benefit of this complication? Are we going to inject the other config on the fly?

Reply:
@aholovko it is there just to have something that returns an interface we can mock in order to bump up test coverage. There are tests that are overriding this supplier.